File: Helpers/src/main/java/com/dg/runnable/RunnableWithArgument.java
package com.dg.runnable;

/**
 * Created by Daniel Cohen Gindi (danielgindi@gmail.com)
 */
public abstract class RunnableWithArgument<T> implements Runnable
{
    private T _runnableArgument = null;

    public void setRunnableArgument(T arg)
    {
        _runnableArgument = arg;
    }
    public T getRunnableArgument()
    {
        return _runnableArgument;
    }

    @Override
    public void run()
    {
        runWithArgument(_runnableArgument);
    }

    public abstract void runWithArgument(T runnableArgument);
}

File: Helpers/src/main/java/com/dg/runnable/RunnableWithArgument.kt
package com.dg.runnable

abstract class RunnableWithArgument<T> : Runnable
{
    var runnableArgument: T? = null

    override fun run()
    {
        runWithArgument(runnableArgument)
    }

    abstract fun runWithArgument(runnableArgument: T?)
}
